vgscollect/src/main/java/com/verygoodsecurity/vgscollect/core/storage/Notifier.kt

```diff
@@ -33,9 +37,13 @@ internal class Notifier: DependencyDispatcher, FieldDependencyObserver {
     private fun notifyRelatedFields(state: VGSFieldState, onDependencyDetected: DependencyHandler) {
         when {
             state.isCardNumberType() -> {
-                val r:Array<Int> = (state.content as? FieldContent.CardNumberContent)?.rangeCVV?.sortedArray()?: arrayOf(3,4)
-                val dependency = Dependency(DependencyType.RANGE, r)
-                onDependencyDetected(FieldType.CVC, dependency)
+                val cardContent = state.content as? FieldContent.CardNumberContent
+                val dependencyRange =
+                    Dependency.range(cardContent?.rangeCVV?.sortedArray() ?: arrayOf(3, 4))
+                onDependencyDetected(FieldType.CVC, dependencyRange)
+                val dependencyCardType =
+                    Dependency.cardType(cardContent?.cardtype ?: CardType.UNKNOWN)
```

Comment:
Pay attention to custom brands. If a developer adds his custom brand, then it won't be an unknown brand (even the CardType will be unknown).

Reply:
@Endorf As I understand we can have a custom brand with an unknown card type?

Reply:
I just highlight the issue. Let's figure it out together.